src/traverseDelegateProperties.js

```diff
@@ -0,0 +1,194 @@
+}
+
+// function pushValueIntoSettings(pathString, settings, value) {
+//   if (!isString(pathString) || !isObject(settings)) {
+//     return undefined;
+//   }
+//
+//   var settingsCopy = objectAssign({}, settings);
+//
+//   var pathSegments = pathString.split('.');
+//   var nextSettingsLevel = settingsCopy; // 2 references now point to "settingsCopy"
+//
+//   // Pull off because this is the ultimate variable that needs to be granted the passed in value
+//   var finalSegment = pathSegments.pop();
+//
+//   // traverse down to the inner-most path
+//   pathSegments.forEach(function (pathSegment, index) {
+//     if (isArrayReference(pathSegment)) {
+//       var nextPathKeyParts = sanitizeArrayKeyAndSplit(pathSegment);
+//       nextSettingsLevel =
+//         nextSettingsLevel[nextPathKeyParts.arrName][nextPathKeyParts.arrIndex];
+//     } else {
+//       nextSettingsLevel = nextSettingsLevel[pathSegment];
+//     }
+//   });
+//
+//   // update the value
+//   if (isArrayReference(finalSegment)) {
+//     var nextPathKeyParts = sanitizeArrayKeyAndSplit(finalSegment);
+//     nextSettingsLevel[nextPathKeyParts.arrName][
+//       nextPathKeyParts.arrIndex
+//     ] = value;
+//   } else {
+//     nextSettingsLevel[finalSegment] = value;
+//   }
+//
+//   return settingsCopy;
+// }
+
+function pushValueIntoSettings(
+  filePathString,
+  settings,
+  decorateWithDynamicHost
+) {
+  // nothing to do
+  if (
+    !isString(filePathString) ||
+    !filePathString.length ||
+    !isObject(settings)
+  ) {
+    return;
+  }
+
+  // get the next key to observe
+  var pathSegments = filePathString.split('.');
+  var settingToReplace = pathSegments.pop();
+  if (!settingToReplace) {
+    var error = new Error();
+    error.code = 'malformed-path-string';
+    throw error;
+  }
+
+  // base case
+  if (settings.hasOwnProperty(settingToReplace)) {
+    var url = settings[settingToReplace];
+    settings[settingToReplace] = decorateWithDynamicHost(url);
+    return;
+  }
+
+  // still more work to do
+  if (pathSegments.length) {
+    var currentKey = pathSegments.shift();
+    // the next layer down will need the setting to replace back on the end
+    var remainingPathString = pathSegments.concat(settingToReplace).join('.');
+
+    if (isArrayReference(currentKey)) {
+      // 'someArrayReference[]' --> 'someArrayReference'
+      currentKey = sanitizeArrayKey(currentKey);
+      var settingsValue = settings[currentKey];
+      if (Array.isArray(settingsValue)) {
+        settingsValue.forEach(function (arrayEntryObject) {
+          return pushValueIntoSettings(
+            remainingPathString,
+            arrayEntryObject,
+            decorateWithDynamicHost
+          );
+        });
+      }
+    } else {
+      pushValueIntoSettings(
+        remainingPathString,
+        settings[currentKey],
+        decorateWithDynamicHost
+      );
+    }
+  }
+}
+
+module.exports = {
+  pluckSettingsValue: pluckSettingsValue,
```

Comment:
I don't see this used anywhere. I think we should remove the code related to this method.

Reply:
Yea. It was on my list. This function was useful back when I had a caching strategy.